### PR TITLE
feat: Always show Pro token input

### PR DESCRIPTION
**Pull request by ashuntu**
Shows the token input field always instead of being in a collapsible section. Also, adds messaging when a valid token or invalid token is added, with an icon.

A small detail I missed when adjusting the copy for errors previously: invalid tokens now show the string "Invalid token" instead of "Token invalid".

<details>
<summary>Screenshot of the default look</summary>

image

</details>

<details>
<summary>Screenshot when an invalid token is entered</summary>

image

</details>

<details>
<summary>Screenshot when a valid token is entered</summary>

image

</details>

Our designs in Figma include functionality that we do not have, so I took some creative liberties in adapting it to the current layout. Let me know if it needs any adjustments.

---

UDENG-5284

### CODE REVIEW

File: gui/packages/ubuntupro/lib/constants.dart [hunk 1]
```diff
@@ -12,3 +14,5 @@ const kVersion = String.fromEnvironment(
   'UP4W_FULL_VERSION',
   defaultValue: 'Dev',
 );
+
+const kConfirmColor = Color(0xFF0E8420);
```
Comment (ashuntu) [inline gui/packages/ubuntupro/lib/constants.dart:18]: Maybe we should look at replacing this custom color with a built-in color in the future. It's one of the only custom colors we use in the app (other than `Color(0xFFE6F8E8)` in one place on the subscription status page).

Comment (CarlosNihelton) [reply gui/packages/ubuntupro/lib/constants.dart:18]: Interestingly this constant is the value of `YaruColors.success`.
https://ubuntu.github.io/yaru.dart/
https://github.com/ubuntu/yaru.dart/blob/main/lib/src/themes/colors.dart#L25-L39

I think we should use the Yaru value deliberately now that's available.

**Review by CarlosNihelton — COMMENTED** [review]
I think your proposal finds a good balance between the stuff we know we need to change and the amount we don't know yet.
One of the key benefits of the recent releases of Yaru is the consistency brought by YaruColors. Thus my questions about whether we're approaching them properly or not.

**Comment by ashuntu** [comment]
I converted the colors to Yaru colors. The status page also uses a modified version of success that is extremely close to the custom value we were using.

File: gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_widgets.dart [hunk 2]
```diff
@@ -61,58 +61,97 @@ class _ProTokenInputFieldState extends State<ProTokenInputField> {
...
+                  decoration: InputDecoration(
+                    hintText: lang.tokenInputHint,
+                    error: _token.errorOrNull?.localize(lang) != null
+                        ? Padding(
+                            padding: const EdgeInsets.only(top: 4),
+                            child: Row(
+                              children: [
+                                Icon(
+                                  Icons.cancel,
+                                  color: YaruColors.of(context).error,
```
Comment (ashuntu) [inline gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_widgets.dart:97]: Is there a preference for `theme.colorScheme` vs `YaruColors.of`?

Comment (CarlosNihelton) [reply gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_widgets.dart:97]: For the error case they would mean the same, but `YaruColors` offers `.link`, `.warning` and `.success` which I don't think `theme.colorScheme` does. So, YaruColors is preferrable when those semantics may change. For ease, let's just stick to `YaruColors`.

**Comment by CarlosNihelton** [comment]
> I converted the colors to Yaru colors. The status page also uses a modified version of `success` that is extremely close to the custom value we were using (is there a built-in version of this I don't know about?).

I suspect if you use the `YaruInfo` widget you'll get the correct colors for free ;)

**Comment by ashuntu** [comment]
> I suspect if you use the `YaruInfo` widget you'll get the correct colors for free ;)

That looks promising, I think I will do that in another PR since its unrelated to the token page.

File: gui/packages/ubuntupro/lib/pages/subscription_status/subscription_status_widgets.dart [hunk 3]
```diff
@@ -47,17 +47,18 @@ class SubscriptionStatus extends StatelessWidget {
         Container(
           padding: const EdgeInsets.all(16.0),
           decoration: BoxDecoration(
-            border: Border.all(color: const Color(0xFF0E8420), width: 1.0),
-            color: const Color(0xFFE6F8E8),
+            border: Border.all(color: theme.colorScheme.success, width: 1.0),
+            color: theme.colorScheme.success
+                .copyWith(lightness: 0.94, saturation: 0.56),
```
Comment (CarlosNihelton) [inline gui/packages/ubuntupro/lib/pages/subscription_status/subscription_status_widgets.dart:52]: AFAIK `.success` is an extension method brought you by yaru.dart. So, there is no benefit in not explicitely depending on YaruColors instead.

**Review by CarlosNihelton — APPROVED** [review]
Good job!

**Pull request merged**